distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Aggregator.enso

```diff
@@ -0,0 +1,149 @@
+            if mid_value <= 1 then value.first.first else
+                if mid_value >= count then value.last.first else
+                    mid = mid_value.floor
+                    output = value.fold_with_key [0, Nothing, Nothing] c->k->v->
+                        new_v = c.first + v
+                        new_s = if c.first.up_to new_v . contains (mid-1) then k else c.second
+                        new_t = if c.first.up_to new_v . contains mid then k else (c.at 2)
+                        [new_v, new_s, new_t]
+                    (output.second + (output.at 2 - output.second) * (mid_value - mid))
+
+    case column of
+        Group_By c _ ->
+            create_aggregator accumulator=(create_closure c col->_->i->(col.at i))
+        Count _ ->
+            create_aggregator initial=0 accumulator=(c->_->c+1)
+        Count_Distinct columns _ ignore_nothing ->
+            resolved = case columns of
+                Vector.Vector _ -> columns.map c->(column.resolve_column table c)
+                _ -> [column.resolve_column table columns]
+            key_maker i = Group_By_Key.key (resolved.map c->(c.at i))
+            accumulator = case ignore_nothing of
+                False-> map->i->(map.insert (key_maker i) 1)
+                True-> map->i->
+                    key = key_maker i
+                    if key.values.all .is_nothing then map else (map.insert key 1)
+            create_aggregator initial=Map.empty accumulator=accumulator finalizer=(v->v.size)
+        Count_Not_Nothing c _ ->
+            accumulator = create_closure c col->count->i->(count + if (col.at i).is_nothing then 0 else 1)
+            create_aggregator initial=0 accumulator=accumulator
+        Count_Nothing c _ ->
+            accumulator = create_closure c col->count->i->(count + if (col.at i).is_nothing then 1 else 0)
+            create_aggregator initial=0 accumulator=accumulator
+        Count_Not_Empty c _ ->
+            accumulator = create_closure c col->count->i->(count + if is_empty (col.at i) then 0 else 1)
+            create_aggregator initial=0 accumulator=accumulator
+        Count_Empty c _ ->
+            accumulator = create_closure c col->count->i->(count + if is_empty (col.at i) then 1 else 0)
+            create_aggregator initial=0 accumulator=accumulator
+        Sum c _ ->
+            accumulator = create_closure c col->total->i->
+                v = col.at i
+                if v.is_nothing then total else
+                    if total.is_nothing then v else total + v
+            create_aggregator accumulator=accumulator
+        Average c _ ->
+            accumulator = create_closure c col->a->i->
+                v = col.at i
+                if v.is_nothing then a else [a.first + 1, a.second + v]
+            finalizer value = if value.first == 0 then Nothing else (value.second / value.first)
+            create_aggregator initial=[0, 0] accumulator=accumulator finalizer=finalizer
```

Comment:
I really like it - IMO now it is so much more readable that the whole logic for each type of computation is in a single place.